### PR TITLE
feat(ColorPicker): Optimize the display of ColorPicker pure panel

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Optimize the display of ColorPicker Pure panel

before:
image

after:
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Optimize the display of ColorPicker Pure panel  demo |
| 🇨🇳 Chinese |  优化拾色器面板示例的展示   |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at c9246d9</samp>

This pull request enhances the `ColorPicker` component by adding a new prop to adjust the popover position and improving the pure panel mode. It also updates the `pure-panel.tsx` demo file to add some padding to the `PureRenderColorPicker` component.

### <samp>🤖 Generated by Copilot at c9246d9</samp>

*  Add `autoAdjustOverflow` prop to `ColorPickerProps` interface to control popover position ([link](https://github.com/ant-design/ant-design/pull/42388/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR49))
*  Set default value of `autoAdjustOverflow` to `true` in `ColorPicker` component ([link](https://github.com/ant-design/ant-design/pull/42388/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR77))
*  Pass `autoAdjustOverflow` prop to `Popover` component inside `ColorPicker` component ([link](https://github.com/ant-design/ant-design/pull/42388/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR124))
*  Modify `PurePanel` component to accept custom props mapper function ([link](https://github.com/ant-design/ant-design/pull/42388/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL164-R176))
*  Use custom props mapper function to set `placement` and `autoAdjustOverflow` props for pure panel ([link](https://github.com/ant-design/ant-design/pull/42388/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL164-R176))
*  Add padding to `PureRenderColorPicker` component in `pure-panel.tsx` demo file ([link](https://github.com/ant-design/ant-design/pull/42388/files?diff=unified&w=0#diff-6dabc44dd1c0580b35bd42a19166f11319bc6ce63628f3065ae0193614b002bdL11-R15))

### CODE REVIEW

**Review by zombieJ — APPROVED** [review]

**Comment by afc163** [comment]
覆盖率掉了。

**Comment by kiner-tang** [comment]
image
看起来没变呀。 @afc163

**Comment by kiner-tang** [comment]
哦，master 掉了，我补一下用例吧

File: components/color-picker/demo/pure-panel.tsx [hunk 1]
```diff
@@ -8,5 +8,9 @@ export default () => {
   const { token } = theme.useToken();
   const [color, setColor] = useState<Color | string>(token.colorPrimary);
 
-  return <PureRenderColorPicker value={color} onChange={setColor} />;
+  return (
+    <div style={{ paddingLeft: 100 }}>
```
Comment (MadCcc) [inline components/color-picker/demo/pure-panel.tsx:12]: 这个建议保留原样吧，pure-panel 的用例保持简单就好

Comment (kiner-tang) [reply components/color-picker/demo/pure-panel.tsx:12]: 但如果保持原样的话，debug 模式经常会挡住其他元素，感觉 pure-panel 不需要自动调整位置呀

Comment (MadCcc) [reply components/color-picker/demo/pure-panel.tsx:12]: 无关的，让 pure-render 不改变位置的是 genPurePanel 最后一个参数，这里就算固定了 placement，但是没有第四个参数处理的话还是会自动调整。所以 placement 不是固定位置的必要参数，保留默认的就好，也方便排查问题。

**Pull request merged**